core/src/Iterator.cpp

```diff
@@ -37,16 +37,26 @@ TimePoint Iterator::parseAndSet(const std::string& startTimeStr, const std::stri
     return startTime;
 }
 
-void Iterator::run()
+int Iterator::run()
 {
+    int return_code = 0;
+
     iterant.start(startTime);
 
     for (auto t = startTime; t < stopTime; t += timestep) {
         TimestepTime tsTime = { t, timestep };
-        iterant.iterate(tsTime);
+        try {
```

Comment:
Why catch the exception at this level, just to pass an integer return value up to the next layer?

Reply:
I did that so that the output buffers would be flushed, the file would be closed, and the model would write an end-of-run restart file, which may contain more interesting information. Not flushing the buffers before crashing is a significant problem in the model's current state.

But this can maybe be done more elegantly, e.g.

```
        } catch (const std::exception& e) {
            interant.stop(tsTime);
            throw;
        }
```

And then catch the exception at a higher level.